sycl/plugins/esimd_emulator/pi_esimd_emulator.hpp

```diff
@@ -82,12 +83,19 @@ struct _pi_device : _pi_object {
 struct _pi_context : _pi_object {
   _pi_context(pi_device ArgDevice) : Device{ArgDevice} {}
 
-  /// One-to-one mapping between Context and Device
+  // One-to-one mapping between Context and Device
   pi_device Device;
 
-  /// Map SVM memory starting address to corresponding
-  /// CmBufferSVM object. CmBufferSVM object is needed to release memory.
+  // Map SVM memory starting address to corresponding
+  // CmBufferSVM object. CmBufferSVM object is needed to release memory.
   std::unordered_map<void *, cm_support::CmBufferSVM *> Addr2CmBufferSVM;
+  // Thread-safe mapping management of Addr2CmBufferSVM
+  std::mutex CmSVMMapMutex;
+
+  // Thread-safe CM-Surface Creation/Deletion
```

Comment:
Please name in the comment a resource(s) (variable(s)) which this lock guards access to

Reply:
Done.